### DISCUSSION TITLE
Water valve support

### DISCUSSION THREAD

**Discussion by Snekatis** [mongoose-os-apps/shelly-homekit#1505] [Ideas]
Is it possible add support for Water valve? So HomeKit will see it as a real Water valve?

![HK](https://user-images.githubusercontent.com/459219/78779622-87990400-799d-11ea-83db-a9235393201c.png)

**Comment by ursusHU** [comment]
And also Garage door & Sprinkler please :)

**Comment by Montardit** [comment]
Yes, please! Sprinkler support!

**Comment by danny1605** [comment]
Please, integrate the irrigation type !

**Comment by lucaspinelli85** [comment]
it would be great for my garden, please integrate it!

**Comment by danny1605** [comment]
There is already new information regarding Water valve support ?

**Comment by timoschilling** [comment]
Basic valve support is now merged (#433), timer support maybe follow.

**Comment by rojer** [comment]
please help test at https://github.com/mongoose-os-apps/shelly-homekit/issues/498

**Comment by Montardit** [comment]
Shellys configured as water valve can't be added to automatizations. If I configure it as as switch it is selectable but if it is configured as a valve not. Not as a irrigation nor as basic valve. I've configured it with a shelly 2.5. They don't show in the accessories/ambient selector list of the automatization.

**Comment by timoschilling** [comment]
@Montardit I can confirm that, but it looks like a bug or missing feature in the home app. I created an irrigation valve with the HomeKit Accessory Simulator (a developer app from Apple) and it behaves the same. Google for that and you will find some more reports about that. But you can use apps like Home+ or Eve to create a irrigation related automatisation, maybe using a scene to start the irrigation from a automatisation.

**Comment by Montardit** [comment]
It's a little weird, I hope they'll fix this. It's strange you cannot 
water your plants regularly if you have an irrigation valve... I've 
tried with Eve and it works fine. Home+ is not free.

Thanks!!!

El 17/3/21 a les 3:50, Timo Schilling ha escrit:

**Comment by danny1605** [comment]
hello, is there anything new regarding the timing of irrigation?

**Comment by blitzi2000** [comment]
maybe adding a new HAP Type in the roller shutter mode so the icon can be see as a water tap/sprinkler in homekit? that would allow opening and closing and automatize it with a time setting or a sensor input, is not the same but could work for a motorized valve with 3 cables.
![cr3-01-wiring-diagram-of-motorized-valve](https://user-images.githubusercontent.com/116883020/198571647-cd64abef-3993-4f0a-828c-50e248fb5d55.png)

**Reply by vizlipuzli** [reply]
Yes, that's exactly what I was trying to do, but during calibration, the valve doesn’t have enough time to fully open or close. I need a way to manually set the time required for the valve to complete its opening and closing. Does anyone know how to do this?
